    @Test
    public void testResolveQualifiedNameFails() throws Exception {
        final Set<String> imports = new HashSet<>();
        imports.add("java.applet.someClass");

        final ClassResolver classResolver = PowerMockito.spy(new ClassResolver(Thread
                .currentThread().getContextClassLoader(), "", imports));

        PowerMockito.doThrow(new ClassNotFoundException("expected exception"))
                .when(classResolver, "safeLoad", any());
        PowerMockito.doReturn(true).when(classResolver, "isLoadable", any());

        try {
            classResolver.resolve("someClass", "");
            fail("IllegalStateException is expected");
        }
        catch (IllegalStateException ex) {
            // exception is expected
            final String expected = "expected exception";
            assertTrue("Invalid exception cause, should be: ClassNotFoundException",
                    ex.getCause() instanceof ClassNotFoundException);
            assertTrue("Invalid exception message, should end with: " + expected,
                    ex.getMessage().endsWith(expected));
        }
    }

